src/Illuminate/Support/Traits/EnumeratesValues.php

```diff
@@ -222,6 +222,29 @@ public function every($key, $operator = null, $value = null)
         return $this->every($this->operatorForWhere(...func_get_args()));
     }
 
+    /**
+     * Determine if all items are equals.
+     *
+     * @param  callable  $callback
+     * @return bool
+     */
+    public function allEquals(callable $callback = null)
+    {
+        if ($this->isEmpty()) {
```

Comment:
Why not just `return $this->unique($callback)->count() < 2;` ?

Reply:
Because performance issues. If we use a large collection, best case is too cost as the best case.